framework/test/unit/node/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -332,9 +333,112 @@ describe('CommitPool', () => {
+			address: getRandomBytes(20),
+			blsPublicKey: getRandomBytes(48),
+			blsSecretKey: getRandomBytes(32),
+		};
+		const validatorInfo2 = {
+			address: getRandomBytes(20),
+			blsPublicKey: getRandomBytes(48),
+			blsSecretKey: getRandomBytes(32),
+		};
+		const certificate1 = computeCertificateFromBlockHeader(blockHeader1);
+		const certificate2 = computeCertificateFromBlockHeader(blockHeader2);
+		let expectedCommit: AggregateCommit;
+		let context: APIContext;
+		const singleCommit1 = {
+			blockID: blockHeader1.id,
+			height: blockHeader1.height,
+			validatorAddress: validatorInfo1.address,
+			certificateSignature: signCertificate(
+				validatorInfo1.blsSecretKey,
+				networkIdentifier,
+				certificate1,
+			),
+		};
+		const singleCommit2 = {
+			blockID: blockHeader2.id,
+			height: blockHeader2.height,
+			validatorAddress: validatorInfo2.address,
+			certificateSignature: signCertificate(
+				validatorInfo2.blsSecretKey,
+				networkIdentifier,
+				certificate2,
+			),
+		};
+		const singleCommits = [singleCommit1, singleCommit2];
+		const validatorKeys = [validatorInfo1.blsPublicKey, validatorInfo2.blsPublicKey];
+		validatorKeys.sort((blsKeyA, blsKeyB) => blsKeyA.compare(blsKeyB));
+		const pubKeySignaturePair1 = {
+			publicKey: validatorInfo1.blsPublicKey,
+			signature: singleCommit1.certificateSignature,
+		};
+		const pubKeySignaturePair2 = {
+			publicKey: validatorInfo2.blsPublicKey,
+			signature: singleCommit2.certificateSignature,
+		};
+		const pubKeySignaturePairs = [pubKeySignaturePair1, pubKeySignaturePair2];
+
+		beforeEach(() => {
+			commitPool = new CommitPool({
+				bftAPI,
+				validatorsAPI,
+				blockTime,
+				network,
+				chain,
+			});
+			context = testing.createTransientAPIContext({});
+			when(validatorsAPI.getValidatorAccount)
+				.calledWith(expect.any(Object), validatorInfo1.address)
+				.mockReturnValue({ blsKey: validatorInfo1.blsPublicKey });
+			when(validatorsAPI.getValidatorAccount)
+				.calledWith(expect.any(Object), validatorInfo2.address)
+				.mockReturnValue({ blsKey: validatorInfo2.blsPublicKey });
+		});
+
+		it('should throw if there are no single commits', async () => {
+			await expect(commitPool.aggregateSingleCommits(context, [])).rejects.toThrow(
+				'No single commit found',
+			);
+		});
+
+		it('should return aggregated commit if there is atleast 1 single commit', async () => {
+			const { aggregationBits, signature: aggregateSignature } = createAggSig(
+				[validatorInfo1.blsPublicKey],
+				[pubKeySignaturePair1],
+			);
+			expectedCommit = { height, aggregationBits, certificateSignature: aggregateSignature };
+			bftAPI.getBFTParameters.mockReturnValue({
+				validators: [{ address: validatorInfo1.address }],
+			});
+
+			await expect(
+				commitPool.aggregateSingleCommits(context, [singleCommit1]),
+			).resolves.toStrictEqual(expectedCommit);
+		});
+
+		it('should return aggregated commit for multiple single commits', async () => {
```

Comment:
maybe check validatorKeys are called in lexicographical order?

Reply:
not sure how to test this.
Also, strangely `createAggSig(validatorKeys,pubKeySignaturePairs)` gives the same output irrespective of order of `validatorKeys` so not sure why are the keys being sorted in lexicographical order ?

Reply:
aggregation bit should be different depending on the order of the keys

Reply:
I tried for multiple fixtures but aggregationBits is not different depending on the order of the keys.
Please look at `L509` in the test where this is tested (fails) `expect(aggregationBits).not.toStrictEqual(aggregationBits2)`